### PR TITLE
Fix big number issues in l2 cancellation creation

### PR DESCRIPTION
**Pull request by calvogenerico**
Parsing of really big numbers was failing in a few places of the new cancellation flow. Changes made:

- Avoid using "number" type in every part of the flow. Instead using bigint or numerical strings.
- Avoid bigints with remix-params-helper. There is a bug there that makes it fail. Instead using simple custom function to extract data.

### CODE REVIEW

File: cli/test/block-explorer-client.spec.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { beforeEach, describe, expect, it } from "vitest";
 import { BlockExplorerClient } from "../src";
 import { ExternalApiError } from "../src/lib/errors";
 
-describe("BlockExplorerClient", () => {
+describe.skip("BlockExplorerClient", () => {
```
Comment (aon) [inline cli/test/block-explorer-client.spec.ts:5]: Should this test suite be skipped?

File: webapp/app/common/basic-schemas.ts [hunk 2]
```diff
@@ -9,3 +9,11 @@ export const hexSchema = hexChars.transform((str) => str as Hex);
...
   .refine((str) => str.length === 42, "Address have to be 20 bytes long")
   .transform((str) => getAddress(str));
+
+export const numericStrSchema = z.coerce
+  .bigint({ message: "Must be a integer number" })
+  .transform((num) => num.toString());
+
+export const nonZeroNumericStrSchema = numericStrSchema.refine((s) => !/^0+$/.test(s), {
+  message: "Cannot be zero.",
+});
```
Comment (aon) [inline webapp/app/common/basic-schemas.ts:19]: Can we do this check in the same way as the other? Transforming to number then back?

File: webapp/app/common/basic-schemas.ts [hunk 3]
```diff
@@ -9,3 +9,11 @@ export const hexSchema = hexChars.transform((str) => str as Hex);
 export const addressSchema = hexChars
   .refine((str) => str.length === 42, "Address have to be 20 bytes long")
   .transform((str) => getAddress(str));
+
+export const numericStrSchema = z.coerce
+  .bigint({ message: "Must be a integer number" })
+  .transform((num) => num.toString());
```
Comment (aon) [inline webapp/app/common/basic-schemas.ts:15]: I'd rename this to `bigintStrSchema` to not confuse it with any future `numericStrSchema` that's actually of number type.

**Review by aon — APPROVED** [review]

File: webapp/app/common/basic-schemas.ts (hunk 2 above)
Comment (calvogenerico) [reply webapp/app/common/basic-schemas.ts]: Done

File: cli/test/block-explorer-client.spec.ts (hunk 1 above)
Comment (calvogenerico) [reply cli/test/block-explorer-client.spec.ts]: No. But they were doin an api call to etherscan and etherscan was down. But It's back online. Removing the sklip

File: webapp/app/common/basic-schemas.ts (hunk 3 above)
Comment (calvogenerico) [reply webapp/app/common/basic-schemas.ts]: Renamed!

**Review by aon — APPROVED** [review]

**Pull request merged**